### PR TITLE
p2p: banscore updates to gui, tests, release notes

### PR DESCRIPTION
**Pull request by jonatack**
This is the third `-banscore` PR in the mini-series described in #19464. See that PR for the intention and reasoning.

- no longer display "Ban Score" in the GUI peers window and add a release note, plus release note fixups per https://github.com/bitcoin/bitcoin/pull/19464#pullrequestreview-447452052
- update tests (`src/test/denialofservice_tests.cpp` and `test/functional/p2p_leak.py`) from banning to discouragement and per https://github.com/bitcoin/bitcoin/pull/19464#issuecomment-658052518

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -303,6 +304,11 @@ issue.
 GUI changes
 -----------
 
+- The GUI Peers window no longer displays a "Ban Score" field. This is part of
+  changes in 0.20.1 and in this release to the handling of misbehaving
+  peers. Refer to "Changes regarding misbehaving peers" in the 0.20.1 release
+  notes for details. (#19510)
```
Comment (jonatack) [inline doc/release-notes.md:310]: s/19510/19512/

**Comment by jnewbery** [comment]
code review ACK 38cc0003812dbc7199bc00f5801f7ddb9ec89e37.

I think we should also remove `-bantime`, `DEFAULT_MISBEHAVING_BANTIME`, `m_default_ban_time` and the default time argument to `Banman::Ban()` now as well. Since `Ban()` can now only be called from the RPC or GUI, the user can provide a bantime on each call (with a default provided by RPC/GUI).

**Comment by jonatack** [comment]
Thanks for reviewing @jnewbery. Updated the second commit to follow up on your suggestion at https://github.com/bitcoin/bitcoin/pull/19464#issuecomment-658052518. 

`git diff 38cc000 fa108d6`

**Comment by jonatack** [comment]
> I think we should also remove `-bantime`, `DEFAULT_MISBEHAVING_BANTIME`, `m_default_ban_time` and the default time argument to `Banman::Ban()` now as well. Since `Ban()` can now only be called from the RPC or GUI, the user can provide a bantime on each call (with a default provided by RPC/GUI).

You know how I love removing code. Will have a look for a follow-up.

**Comment by jnewbery** [comment]
ACK fa108d6a757838225179a8df942cfb6d99c98c90

**Comment by laanwj** [comment]
ACK fa108d6a757838225179a8df942cfb6d99c98c90
Did code review and tested the GUI change.

**Pull request merged**